tests/unit/s2n_early_data_test.c

```diff
@@ -312,5 +326,50 @@ int main(int argc, char **argv)
         EXPECT_EQUAL(psk->early_data_config.context.allocated, 0);
     }
 
+    /* Test s2n_early_data_config_clone */
+    {
+        const uint8_t test_bad_value[] = "wrong";
+        const uint8_t test_apln[] = "protocol";
+        const uint8_t test_context[] = "context";
+        const uint32_t test_max_early_data = 10;
+        const struct s2n_cipher_suite *test_cipher_suite = &s2n_tls13_chacha20_poly1305_sha256;
+
+        for (size_t called_directly = 0; called_directly < 2; called_directly++) {
+            struct s2n_psk *original = s2n_external_psk_new();
+            EXPECT_NOT_NULL(original);
+            EXPECT_SUCCESS(s2n_psk_configure_early_data(original, test_max_early_data,
+                    test_cipher_suite->iana_value[0], test_cipher_suite->iana_value[1]));
+            EXPECT_SUCCESS(s2n_psk_set_application_protocol(original, test_apln, sizeof(test_apln)));
+            EXPECT_SUCCESS(s2n_psk_set_context(original, test_context, sizeof(test_context)));
+
+            DEFER_CLEANUP(struct s2n_psk *clone = s2n_external_psk_new(), s2n_psk_free);
+            EXPECT_SUCCESS(s2n_psk_set_application_protocol(clone, test_bad_value, sizeof(test_bad_value)));
+            EXPECT_NOT_NULL(clone);
+
+            if (called_directly) {
+                EXPECT_OK(s2n_early_data_config_clone(clone, &original->early_data_config));
+            } else {
+                EXPECT_SUCCESS(s2n_psk_set_identity(original, test_bad_value, sizeof(test_bad_value)));
+                EXPECT_SUCCESS(s2n_psk_set_secret(original, test_bad_value, sizeof(test_bad_value)));
+                EXPECT_OK(s2n_psk_clone(clone, original));
+            }
+
+            /* Free the original to ensure they share no memory */
+            EXPECT_SUCCESS(s2n_psk_free(&original));
+
+            /* existing alpn is replaced by original's alpn */
+            EXPECT_EQUAL(clone->early_data_config.application_protocol.size, sizeof(test_apln));
+            EXPECT_BYTEARRAY_EQUAL(clone->early_data_config.application_protocol.data, test_apln, sizeof(test_apln));
+
+            /* new context is allocated for original's context */
+            EXPECT_EQUAL(clone->early_data_config.context.size, sizeof(test_context));
+            EXPECT_BYTEARRAY_EQUAL(clone->early_data_config.context.data, test_context, sizeof(test_context));
+
+            /* other values are copied */
+            EXPECT_EQUAL(clone->early_data_config.max_early_data_size, test_max_early_data);
+            EXPECT_EQUAL(clone->early_data_config.cipher_suite, test_cipher_suite);
```

Comment:
We should also assert that the `protocol_version` was copied from original to new.